DirectProgramming/DPC++FPGA/ReferenceDesigns/decompress/src/snappy/snappy_reader.hpp

```diff
@@ -175,13 +175,14 @@ unsigned SnappyReader(unsigned in_count) {
   while (bytes_processed_in_range) {
     // grab new bytes if there is space
     if (byte_stream.Space() >= literals_per_cycle) {
+
       bool valid_read;
       auto pipe_data = InPipe::read(valid_read);
       if (valid_read) {
         byte_stream.template Write(pipe_data);
         data_read += literals_per_cycle;
         all_data_read = all_data_read_next;
-        all_data_read_next = data_read >= in_count - 2 * literals_per_cycle;
+        all_data_read_next = data_read >= in_count - literals_per_cycle;

```

Comment:
What happened to the 2 x ?

Reply:
I don't think that it was correct.
I've printed the values in the program to check that we indeed read the entire program with this new value and this is the case.
This is actually what was causing the hang on linux.
It was assuming the read was complete before it was actually done.
Depending on if there was something to read from the pipe at the next iteration or not, the program either continued to read beyond what it thought was `all_data_read` and complete successfully, or stop at `all_data_read` and hang.